src/components/EventAdd/index.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { castDraft } from 'immer';
+
+import Button from '../Button';
+import { classes, getRandomColor } from '../../utils/misc';
+import { DAYS } from '../../constants';
+import { ScheduleContext } from '../../contexts';
+
+import './stylesheet.scss';
+
+export type EventAddProps = {
+  className?: string;
+};
+
+export default function EventAdd({
+  className,
+}: EventAddProps): React.ReactElement {
```

Comment:
Let's add an option to prefill event info, since we'll be using this form for editing events as well. When an event's id is present, no need to regenerate id.